src/IIIF/IIIF/Serialisation/Deserialisation/ServiceConverter.cs

```diff
@@ -15,17 +15,37 @@ public class ServiceConverter : ReadOnlyConverter<IService>
         bool hasExistingValue, JsonSerializer serializer)
     {
         var jsonObject = JObject.Load(reader);
+        
+        var atType = jsonObject["@type"];
+
+        var result = ImageServiceTwoResult(serializer, atType, jsonObject);
+        if (result != null) return result;
 
-        var service = IdentifyConcreteType(jsonObject);
+        var service = IdentifyConcreteType(jsonObject, atType);
 
         serializer.Populate(jsonObject.CreateReader(), service);
         return service;
     }
 
-    private static IService? IdentifyConcreteType(JObject jsonObject)
+    private IService? ImageServiceTwoResult(JsonSerializer serializer, JToken? atType, JObject jsonObject)
+    {
+        var typeValue = atType?.Value<string>();
+        if (typeValue is "iiif:Image" or nameof(ImageService2))
+        {
+            // avoids stack overflow exceptions from recursively calling the service converter

```

Comment:
This is what `CreateCopy()` is for, without changing the current `JsonSerializer` as there can be unexpected side effects when doing that. Instead it creates a copy of the current serializer excluding a specific one

```cs
var x = serializer.CreateCopy(converter => converter is not ServiceConverter);
var result = jsonObject.ToObject(typeof(ImageService2), x);
```